### DISCUSSION TITLE
Cake sevenzip: how to ignore full path in zip file

### DISCUSSION THREAD

**Discussion by ksvenkatesan77** [cake-build/cake#4104] [Extension Q&A]
i tried using sevenzip in a cake file. 
alas, it is added the file with full path. 
![image](https://user-images.githubusercontent.com/110341366/210064836-af067bc1-8417-4c1d-9ed8-a28c15688ca1.png)

i wanted to ignore the full path and add only the files.  are there any ways to do it easily? my code snippet is
`     _context.SevenZip(m => m
           .InUpdateMode()
           .WithArchive(_context.File(fileUpdate.Destination))
           .WithFiles(_context.File(fileUpdate.Source))
           .WithPassword(password));
`

**Comment by nils-a** [comment]
@ksvenkatesan77 this behavior is standard for 7zip. 

All files are added either with their relative or absolute path, as seen from the working directory.
(You can switch between the two modes by adding the [`-spf`/`-spf2`](https://sevenzip.osdn.jp/chm/cmdline/switches/spf.htm) switches in 7zip. Have a look at [`WithFullyQualifiedFilePaths<T>(T, bool)`](https://cake-contrib.github.io/Cake.7zip/api/Cake.SevenZip.Builder/SwitchFullyQualifiedFilePathsBuilder/BFC12033) of the [SwitchFullyQualifiedFilePathsBuilder](https://cake-contrib.github.io/Cake.7zip/api/Cake.SevenZip.Builder/SwitchFullyQualifiedFilePathsBuilder/) on how to do this in *Cake.7zip*)

If you do not want to have any path added to the output, you'd need to specify the working directory, using [`-w`](https://sevenzip.osdn.jp/chm/cmdline/switches/working_dir.htm) or [WithWorkingDirectory<T>(T, DirectoryPath)](https://cake-contrib.github.io/Cake.7zip/api/Cake.SevenZip.Builder/SwitchWorkingDirectoryBuilder/5CC3F36F) of the [SwitchWorkingDirectoryBuilder](https://cake-contrib.github.io/Cake.7zip/api/Cake.SevenZip.Builder/SwitchWorkingDirectoryBuilder/).

So, your code should be look something like this:

```cs
var source = _context.File(fileUpdate.Source).Path;
SevenZip(m => m
      .InUpdateMode()
      .WithArchive(_context.File(fileUpdate.Destination))
      .WithWorkingDirectory(source.GetDirectory())
      .WithFiles(source.GetFilename())
      .WithPassword(password));
```

(Have not tested it, though.)

**Reply by ksvenkatesan77** [reply]
Dear @nils-a 
Thank you very much for your reply. i will test this one and keep you posted

**Reply by ksvenkatesan77** [reply]
Dear @nils-a 

I tried your suggestions. unfortunately, it didn't work as expected. i get same result than earlier. :(
Are there any alternatives?

**Reply by ksvenkatesan77** [reply]
@nils-a 
Managed to solve this issue following 7z documentation.. 2nd highlighted pointed is the solution
![image](https://user-images.githubusercontent.com/110341366/216591216-786f8ecf-4f2b-4a61-8de8-0c89390b9c83.png)

in the filename: i give "./" but i see that sevenzip library removed that while zipping.. not sure if this is an issue in sevenzip library.. But, we tricked by giving additional "./" prefix which solved my problem.

Thank you for your support so far,

**Reply by nils-a** [reply]
I got to admit, I'm not much of a 7zip-pro. 
I always fumble with the commandline until I get the parameters right. 😬 (And then I re-apply those parameters to Cake.7zip)

Good that you solved it!

BTW, you can also call cake with the `--settings_showprocesscommandline=true` parameter, to see which commandline Cake.7zip creates.
